pyb4model/pyb4model.py

```diff
@@ -24,10 +30,37 @@ def missing_val(df, method):
     1  1  5  6
     2  7  8  9
     """
+    
+    # tests
+    
+    if method not in ['delete', 'mean', 'knn']:
+        raise ValueError('valid methods only include "delete", "mean", and "regression"')
+    
+    if not isinstance(df, pd.DataFrame) and not isinstance(df, np.ndarray) and not isinstance(df, pd.Series):
+        raise TypeError('df must be a dataframe, series, or array')
+        
+    if df.empty: # edge case
+        raise ValueError('dataframe cannot be empty')
+        
+    for i in range(len(df.columns)): # edge case
+        if df.iloc[:,i].isnull().sum()==len(df):
+            raise ValueError('dataframe cannot columns with all NaN values')
+    
+    # function
+    
+    if method=='delete':
+        df = df.dropna()
+    
+    if method=='mean':
+        df = df.fillna(df.mean())
+        
+    if method=='knn':
```

Comment:
This is not an error, just an suggestion for something we can implement in the future if we have time: we could accept argument that is case insensitive. 
```suggestion
    if method.lower()=='delete':
        df = df.dropna()
    
    if method.lower()=='mean':
        df = df.fillna(df.mean())
        
    if method.lower()=='knn':
```

Reply:
great idea! i will definitely implement this !